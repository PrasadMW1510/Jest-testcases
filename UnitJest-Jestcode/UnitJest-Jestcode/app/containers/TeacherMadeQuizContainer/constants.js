/*
 *
 * TeacherMadeQuizContainer constants
 *
 */

export const FORM_TEACHER_MADE_QUIZ = 'form_teacher_made_quiz';

export const META_DATA_QUIZ = 'meta_data_quiz';

export const GET_INSTALLEDQUIZDATA_REQUEST =
  'AdvancedSearchContainer/get_installedquizdata_request';
export const GET_INSTALLEDQUIZDATA_REQUEST_SUCCESS =
  'AdvancedSearchContainer/get_installedquizdata_request_success';
export const GET_INSTALLEDQUIZDATA_REQUEST_FAILURE =
  'AdvancedSearchContainer/get_installedquizdata_request_failure';

export const POST_ADD_TEACHERMADEQUIZ_REQUEST =
  'TeacherMadeQuizContainer/form_teacher_made_quiz_request';
export const POST_ADD_TEACHERMADEQUIZ_REQUEST_FAILURE =
  'TeacherMadeQuizContainer/form_teacher_made_quiz_failure';
export const POST_ADD_TEACHERMADEQUIZ_REQUEST_SUCCESS =
  'TeacherMadeQuizContainer/form_teacher_made_quiz_success';

export const GET_ADD_TEACHERMADEQUIZ_DETAILS_REQUEST =
  'TeacherMadeQuizContainer/form_teacher_made_quiz_details_request';
export const GET_ADD_TEACHERMADEQUIZ_DETAILS_REQUEST_FAILURE =
  'TeacherMadeQuizContainer/form_teacher_made_quiz_details_failure';
export const GET_ADD_TEACHERMADEQUIZ_DETAILS_REQUEST_SUCCESS =
  'TeacherMadeQuizContainer/form_teacher_made_quiz_details_success';

export const DELETE_TEACHERMADEQUIZ_REQUEST =
  'TeacherMadeQuizContainer/delete_teacher_made_quiz_details_srequest';
export const DELETE_TEACHERMADEQUIZ_REQUEST_FAILURE =
  'TeacherMadeQuizContainer/delete_teacher_made_quiz_details_failure';
export const DELETE_TEACHERMADEQUIZ_REQUEST_SUCCESS =
  'TeacherMadeQuizContainer/delete_teacher_made_quiz_details_success';
export const DELETE_TEACHERMADEQUIZ_OLD_QUESTIONS =
  'TeacherMadeQuizContainer/delete_teacher_made_old_questions';

export const FORM_FIELD_NAMES = {
  TeacherMadeQuizAuthorTitle: 'teacherMadeQuizTitle',
  teacherMadeQuizTitle: 'teacherMadeQuizTitle',
  TeacherMadeQuizAuthorFirstName: 'teacherMadeQuizAuthorFirstName',
  TeacherMadeQuizAuthorLastName: 'teacherMadeQuizAuthorLastName',
  TeacherMadeQuizNumberOfWords: 'teacherMadeQuizNumberOfWords',
  TeacherMadequizLexileLevel: 'teacherMadequizLexileLevel',
  TeacherMadequizReadingLevel: 'teacherMadequizReadingLevel',
  TeacherMadequizPoints: 'teacherMadequizPoints',
  TeacherMadequizGRL: 'teacherMadequizGRL',
  TeacherMadequizType: 'teacherMadequizType',
  EditQuizCollectionNames: 'editQuizCollectionNames',
};
