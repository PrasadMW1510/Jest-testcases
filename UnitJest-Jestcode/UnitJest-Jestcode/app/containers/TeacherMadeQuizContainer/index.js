/**
 *
 * TeacherMadeQuizContainer
 *
 */

export { default } from './TeacherMadeQuizContainer';
