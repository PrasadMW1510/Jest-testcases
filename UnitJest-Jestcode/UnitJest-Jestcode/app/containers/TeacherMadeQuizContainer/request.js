import axios from 'axios';
import API, { xmlToJSON, getBaseUrlWithoutSlms } from 'utils/request';

export const getInstalledQuizData = sessionId =>
  axios
    .get(`${getBaseUrlWithoutSlms()}/src/SrcQuizManager`, {
      params: {
        command: 'get_teacher_made_quizzes',
        sid: sessionId,
      },
    })
    .then(response => xmlToJSON(response.data));

export const postTeacherMadeQuizRequest = (sessionId, messageObject) =>
  API.post(`${getBaseUrlWithoutSlms()}/src/SrcQuizManager`, messageObject.quizObject, {
    params: {
      command: 'set_quiz',
      sid: sessionId,
    },
  }).then(response => response);

export const getQuizDetailData = (sessionId, quizObj) =>
  API.post(`${getBaseUrlWithoutSlms()}/src/SrcQuizManager`, quizObj, {
    params: {
      command: 'get_quiz_with_questions',
      sid: sessionId,
    },
  }).then(response => response);

export const deleteQuizDetailData = (sessionId, quizObjnew) =>
  API.post(`${getBaseUrlWithoutSlms()}/src/SrcQuizManager`, quizObjnew, {
    params: {
      command: 'delete_teacher_made_quiz',
      sid: sessionId,
    },
  }).then(response => response);
