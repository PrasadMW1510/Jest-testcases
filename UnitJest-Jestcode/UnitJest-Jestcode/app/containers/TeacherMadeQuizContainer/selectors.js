import { createSelector } from 'reselect';

const selectTeacherMadeQuizContainerDomain = state => state.get('teacherMadeQuizContainer');
/**
 * Default selector used by TeacherMadeQuizContainer
 */
const makeSelectTeacherMadeQuizContainer = () =>
  createSelector(selectTeacherMadeQuizContainerDomain, substate => substate.toJS());

export default makeSelectTeacherMadeQuizContainer;
export { selectTeacherMadeQuizContainerDomain };
