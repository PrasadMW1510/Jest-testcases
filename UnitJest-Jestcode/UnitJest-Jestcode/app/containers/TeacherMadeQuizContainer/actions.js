/*
 *
 * TeacherMadeQuizContainer actions
 *
 */

import * as Constants from './constants';

export function getInstalledQuizDataRequest() {
  return {
    type: Constants.GET_INSTALLEDQUIZDATA_REQUEST,
  };
}

export function getInstalledQuizDataRequestSuccess(installedQuizData = []) {
  return {
    type: Constants.GET_INSTALLEDQUIZDATA_REQUEST_SUCCESS,
    installedQuizData,
  };
}

export function getInstalledQuizDataRequestFailure(error) {
  return {
    type: Constants.GET_INSTALLEDQUIZDATA_REQUEST_FAILURE,
    error,
  };
}

export function postTeacherMadeQuizRequest(quizObject) {
  return {
    type: Constants.POST_ADD_TEACHERMADEQUIZ_REQUEST,
    quizObject,
  };
}

export function postTeacherMadeQuizRequestSuccess() {
  return {
    type: Constants.POST_ADD_TEACHERMADEQUIZ_REQUEST_SUCCESS,
  };
}

export function postTeacherMadeQuizRequestFailure(error) {
  return {
    type: Constants.POST_ADD_TEACHERMADEQUIZ_REQUEST_FAILURE,
    error,
  };
}
export function getTeacherMadeQuizDetailsRequest(key) {
  return {
    type: Constants.GET_ADD_TEACHERMADEQUIZ_DETAILS_REQUEST,
    key,
  };
}

export function getInstalledQuizDetailDataRequestSuccess(quizQuestions = []) {
  return {
    type: Constants.GET_ADD_TEACHERMADEQUIZ_DETAILS_REQUEST_SUCCESS,
    quizQuestions,
  };
}
export function deleteQuiz(quizId) {
  return {
    type: Constants.DELETE_TEACHERMADEQUIZ_REQUEST,
    quizId,
  };
}

export function deleteTeacherMadeQuizRequestFailure(error) {
  return {
    type: Constants.DELETE_TEACHERMADEQUIZ_REQUEST_FAILURE,
    error,
  };
}

export function deleteTeacherMadeQuizRequestSuccess(delSuccess = []) {
  return {
    type: Constants.DELETE_TEACHERMADEQUIZ_REQUEST_SUCCESS,
    delSuccess,
  };
}

export function clearTeacherMadeOldQuestions(delSuccess = []) {
  return {
    type: Constants.DELETE_TEACHERMADEQUIZ_OLD_QUESTIONS,
    delSuccess,
  };
}
