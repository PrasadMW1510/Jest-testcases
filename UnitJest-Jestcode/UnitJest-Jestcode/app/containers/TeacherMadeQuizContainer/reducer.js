/*
 *
 * TeacherMadeQuizContainer reducer
 *
 */
import { fromJS } from 'immutable';
import * as Constants from './constants';

const initialState = fromJS({
  error: false,
  installedQuizData: [],
  installedQuizQuestionList: [],
});

function teacherMadeQuizContainerReducer(state = initialState, action) {
  switch (action.type) {
    case Constants.GET_INSTALLEDQUIZDATA_REQUEST_SUCCESS:
      return state.set(
        'installedQuizData',
        fromJS(action.installedQuizData.output.output_data[0].GetTeacherMadeQuizzesResp[0].Book)
      );
    case Constants.GET_ADD_TEACHERMADEQUIZ_DETAILS_REQUEST_SUCCESS:
      return state.set(
        'installedQuizQuestionList',
        fromJS(action.quizQuestions.output_data[0].GetQuizWithQuestionsResp[0].Book[0])
      );
    case Constants.DELETE_TEACHERMADEQUIZ_OLD_QUESTIONS:
      return state.set('installedQuizQuestionList', []);
    default:
      return state;
  }
}

export default teacherMadeQuizContainerReducer;
