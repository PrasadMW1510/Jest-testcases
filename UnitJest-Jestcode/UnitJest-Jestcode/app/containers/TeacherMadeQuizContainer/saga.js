import { call, all, put, takeLatest, select } from 'redux-saga/effects';
import * as Constants from './constants';
import * as Request from './request';
import * as Actions from './actions';
import { makeSelectProfileSessionId } from '../App/selectors';

export function* getInstalledQuizDataRequest() {
  try {
    const sessionId = yield select(makeSelectProfileSessionId());
    const getInstalledQuizData = yield call(Request.getInstalledQuizData, sessionId);
    yield put(Actions.getInstalledQuizDataRequestSuccess(getInstalledQuizData));
  } catch (err) {
    yield put(Actions.getInstalledQuizDataRequestFailure(err));
  }
}

export function* postTeacherMadeQuizRequest(postData) {
  try {
    const sessionId = yield select(makeSelectProfileSessionId());

    const postQuizData = yield call(Request.postTeacherMadeQuizRequest, sessionId, postData);
    if (
      postQuizData &&
      postQuizData.output_data[0] &&
      postQuizData.output_data[0].SetQuizResp[0] &&
      postQuizData.output_data[0].SetQuizResp[0].QuizID[0]
    ) {
      yield put(Actions.getInstalledQuizDataRequest());
    } else {
      yield put(Actions.postTeacherMadeQuizRequestFailure({ error: 'post failed' }));
    }
  } catch (err) {
    yield put(Actions.postTeacherMadeQuizRequestFailure(err));
  }
}
export function* postTeacherMadeQuizDetailRequest(value) {
  try {
    const sessionId = yield select(makeSelectProfileSessionId());
    const quizId = { GetQuizWithQuestionsReq: { QuizID: value.key[0] } };
    const getQuizDetail = yield call(Request.getQuizDetailData, sessionId, quizId);
    yield put(Actions.getInstalledQuizDetailDataRequestSuccess(getQuizDetail));
  } catch (err) {
    yield put(Actions.getInstalledQuizDataRequestFailure(err));
  }
}

export function* deleteTeacherMadeQuizRequest(quizid) {
  try {
    const sessionId = yield select(makeSelectProfileSessionId());
    const quizdelObj = `<DeleteTeacherMadeQuizReq><Delete><QuizID>${
      quizid.quizId[0]
    }</QuizID></Delete></DeleteTeacherMadeQuizReq>`;
    const deleteQuizDetail = yield call(Request.deleteQuizDetailData, sessionId, quizdelObj);
    if (
      deleteQuizDetail &&
      deleteQuizDetail.output_data[0] &&
      deleteQuizDetail.output_data[0].DeleteTeacherMadeQuizResp[0] &&
      deleteQuizDetail.output_data[0].DeleteTeacherMadeQuizResp[0].result[0] &&
      deleteQuizDetail.output_data[0].DeleteTeacherMadeQuizResp[0].result[0] === 'succeed'
    ) {
      yield put(Actions.getInstalledQuizDataRequest());
    } else {
      yield put(Actions.deleteTeacherMadeQuizRequestFailure({ error: 'Quiz not deleted' }));
    }
    yield put(Actions.deleteTeacherMadeQuizRequestSuccess(deleteQuizDetail));
  } catch (err) {
    yield put(Actions.deleteTeacherMadeQuizRequestFailure(err));
  }
}

export default function* defaultSaga() {
  yield all([
    takeLatest(Constants.GET_INSTALLEDQUIZDATA_REQUEST, getInstalledQuizDataRequest),
    takeLatest(Constants.POST_ADD_TEACHERMADEQUIZ_REQUEST, postTeacherMadeQuizRequest),
    takeLatest(Constants.GET_ADD_TEACHERMADEQUIZ_DETAILS_REQUEST, postTeacherMadeQuizDetailRequest),
    takeLatest(Constants.DELETE_TEACHERMADEQUIZ_REQUEST, deleteTeacherMadeQuizRequest),
  ]);
}
