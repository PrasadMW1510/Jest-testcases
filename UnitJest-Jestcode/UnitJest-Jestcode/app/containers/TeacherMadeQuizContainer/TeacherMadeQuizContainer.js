/**
 *
 * TeacherMadeQuizContainer
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import TeacherMadeQuiz from 'components/TeacherMadeQuiz';
import { fromJS } from 'immutable';
import { hideModal } from 'containers/ModalController/actions';
import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import { makeSelectProfileUserType } from 'containers/App/selectors';
import reducer from './reducer';
import saga from './saga';
import {
  getInstalledQuizDataRequest,
  postTeacherMadeQuizRequest,
  getTeacherMadeQuizDetailsRequest,
  deleteQuiz,
  clearTeacherMadeOldQuestions,
} from './actions';
import makeSelectTeacherMadeQuizContainer from './selectors';

export class TeacherMadeQuizContainer extends React.Component {
  componentWillMount() {
    this.props.getInstalledQuizDataRequest();
  }

  render() {
    const { hideModal: hideModalProp } = this.props;
    return (
      <div>
        <TeacherMadeQuiz
          isOpen
          handleCancel={hideModalProp}
          installedQuizData={this.props.teacherMadeQuizContainer.installedQuizData}
          profileUserType={this.props.profileUserType}
          {...this.props}
        />
      </div>
    );
  }
}

TeacherMadeQuizContainer.defaultProps = {
  teacherMadeQuizContainer: fromJS({}),
};

TeacherMadeQuizContainer.propTypes = {
  teacherMadeQuizContainer: PropTypes.object.isRequired,
  getInstalledQuizDataRequest: PropTypes.func.isRequired,
  hideModal: PropTypes.func,
  profileUserType: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  teacherMadeQuizContainer: makeSelectTeacherMadeQuizContainer(),
  profileUserType: makeSelectProfileUserType(),
});

export function mapDispatchToProps(dispatch) {
  return {
    getInstalledQuizDataRequest: () => dispatch(getInstalledQuizDataRequest()),
    handleCancel: () => dispatch(hideModal()),
    handleSave: values => dispatch(postTeacherMadeQuizRequest(values)),
    loadQuizDetails: values => dispatch(getTeacherMadeQuizDetailsRequest(values)),
    deleteQuiz: values => dispatch(deleteQuiz(values)),
    clearTeacherMadeOldQuestions: () => dispatch(clearTeacherMadeOldQuestions()),
  };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);
const withReducer = injectReducer({ key: 'teacherMadeQuizContainer', reducer });
const withSaga = injectSaga({ key: 'teacherMadeQuizContainer', saga });
export default compose(withReducer, withSaga, withConnect)(TeacherMadeQuizContainer);
