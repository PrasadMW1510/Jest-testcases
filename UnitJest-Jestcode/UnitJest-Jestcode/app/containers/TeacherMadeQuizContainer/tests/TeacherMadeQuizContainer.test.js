import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { TeacherMadeQuizContainer, mapDispatchToProps } from '../TeacherMadeQuizContainer';

describe('Teacher Made Quiz Container', () => {
  let wrapper = null;
  const mockgetInstalledQuizDataRequest = jest.fn();
  const mockpostTeacherMadeQuizRequest = jest.fn();
  const mockgetTeacherMadeQuizDetailsRequest = jest.fn();
  const mockdeleteQuiz = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <TeacherMadeQuizContainer
        getInstalledQuizDataRequest={mockgetInstalledQuizDataRequest}
        postTeacherMadeQuizRequest={mockpostTeacherMadeQuizRequest}
        getTeacherMadeQuizDetailsRequest={mockgetTeacherMadeQuizDetailsRequest}
        deleteQuiz={mockdeleteQuiz}
        handleCancel={jest.fn()}
        handleSave={jest.fn()}
        installedQuizData={[]}
        loadQuizDetails={jest.fn()}
        hideModal={jest.fn()}
      />
    );
  });
  it('Should render TeacherMadeQuizContainer', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('map Dispatch To Props', () => {
    const mockObject = jest.fn;
    const mapDispatchToPropscomp = mapDispatchToProps(mockObject);
    expect(mapDispatchToPropscomp.handleCancel().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.handleSave().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.getInstalledQuizDataRequest().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.loadQuizDetails().mock.calls.length).toBe(0);
    expect(mapDispatchToPropscomp.deleteQuiz().mock.calls.length).toBe(0);
  });
});
