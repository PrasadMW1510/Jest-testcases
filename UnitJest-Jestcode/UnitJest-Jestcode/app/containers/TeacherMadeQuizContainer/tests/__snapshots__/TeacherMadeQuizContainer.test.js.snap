// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teacher Made Quiz Container Should render TeacherMadeQuizContainer 1`] = `
<div>
  <TeacherMadeQuiz
    deleteQuiz={[Function]}
    getInstalledQuizDataRequest={[Function]}
    getTeacherMadeQuizDetailsRequest={[Function]}
    handleCancel={[Function]}
    handleSave={[Function]}
    hideModal={[Function]}
    installedQuizData={Array []}
    isOpen={true}
    loadQuizDetails={[Function]}
    postTeacherMadeQuizRequest={[Function]}
    teacherMadeQuizContainer={
      Immutable.Map {
      }
    }
  />
</div>
`;
